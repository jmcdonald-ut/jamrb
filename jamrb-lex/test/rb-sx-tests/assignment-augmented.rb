a = 9
puts "a: 9 == #{a}"

b = 3
b += a
puts "b: 12 == #{b}"

c = 4
c *= 3
puts "c: 12 == #{c}"

d = 49
d /= 7
puts "d: 7 == #{d}"

e = 5
e **= 2
puts "e: 25 == #{e}"

f = 100
f <<= 1
puts "f: 200 == #{f}"

g = 100
g >>= 1
puts "g: 50 == #{g}"

h = 7
h ^= 3
puts "h: 4 == #{h}"

i = 30
i %= 4
puts "i: 2 == #{i}"

j = 28
j -= 4
puts "j: 24 == #{j}"

k = 3
k |= 9
puts "k: 11 == #{k}"

l = false
l ||= 24
puts "l: 24 == #{l}"

m = true
m &&= 24
puts "m: 24 == #{m}"
