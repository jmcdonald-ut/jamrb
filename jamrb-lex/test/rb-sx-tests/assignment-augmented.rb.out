((Token ((1 0) on_ident "a"))
 (Token ((1 1) on_sp " "))
 (Token ((1 2) on_op "="))
 (Token ((1 3) on_sp " "))
 (Token ((1 4) on_int "9"))
 (Token ((1 5) on_nl "\n"))
 (Token ((2 0) on_ident "puts"))
 (Token ((2 4) on_sp " "))
 (Token ((2 5) on_tstring_beg "\""))
 (Token ((2 6) on_tstring_content "a: 9 == "))
 (Token ((2 14) on_embexpr_beg "\\#{"))
 (Token ((2 16) on_ident "a"))
 (Token ((2 17) on_embexpr_end "}"))
 (Token ((2 18) on_tstring_end "\""))
 (Token ((2 19) on_nl "\n"))
 (Token ((3 0) on_ignored_nl "\n"))
 (Token ((4 0) on_ident "b"))
 (Token ((4 1) on_sp " "))
 (Token ((4 2) on_op "="))
 (Token ((4 3) on_sp " "))
 (Token ((4 4) on_int "3"))
 (Token ((4 5) on_nl "\n"))
 (Token ((5 0) on_ident "b"))
 (Token ((5 1) on_sp " "))
 (Token ((5 2) on_op "+="))
 (Token ((5 4) on_sp " "))
 (Token ((5 5) on_ident "a"))
 (Token ((5 6) on_nl "\n"))
 (Token ((6 0) on_ident "puts"))
 (Token ((6 4) on_sp " "))
 (Token ((6 5) on_tstring_beg "\""))
 (Token ((6 6) on_tstring_content "b: 12 == "))
 (Token ((6 15) on_embexpr_beg "\\#{"))
 (Token ((6 17) on_ident "b"))
 (Token ((6 18) on_embexpr_end "}"))
 (Token ((6 19) on_tstring_end "\""))
 (Token ((6 20) on_nl "\n"))
 (Token ((7 0) on_ignored_nl "\n"))
 (Token ((8 0) on_ident "c"))
 (Token ((8 1) on_sp " "))
 (Token ((8 2) on_op "="))
 (Token ((8 3) on_sp " "))
 (Token ((8 4) on_int "4"))
 (Token ((8 5) on_nl "\n"))
 (Token ((9 0) on_ident "c"))
 (Token ((9 1) on_sp " "))
 (Token ((9 2) on_op "*="))
 (Token ((9 4) on_sp " "))
 (Token ((9 5) on_int "3"))
 (Token ((9 6) on_nl "\n"))
 (Token ((10 0) on_ident "puts"))
 (Token ((10 4) on_sp " "))
 (Token ((10 5) on_tstring_beg "\""))
 (Token ((10 6) on_tstring_content "c: 12 == "))
 (Token ((10 15) on_embexpr_beg "\\#{"))
 (Token ((10 17) on_ident "c"))
 (Token ((10 18) on_embexpr_end "}"))
 (Token ((10 19) on_tstring_end "\""))
 (Token ((10 20) on_nl "\n"))
 (Token ((11 0) on_ignored_nl "\n"))
 (Token ((12 0) on_ident "d"))
 (Token ((12 1) on_sp " "))
 (Token ((12 2) on_op "="))
 (Token ((12 3) on_sp " "))
 (Token ((12 4) on_int "49"))
 (Token ((12 6) on_nl "\n"))
 (Token ((13 0) on_ident "d"))
 (Token ((13 1) on_sp " "))
 (Token ((13 2) on_op "/="))
 (Token ((13 4) on_sp " "))
 (Token ((13 5) on_int "7"))
 (Token ((13 6) on_nl "\n"))
 (Token ((14 0) on_ident "puts"))
 (Token ((14 4) on_sp " "))
 (Token ((14 5) on_tstring_beg "\""))
 (Token ((14 6) on_tstring_content "d: 7 == "))
 (Token ((14 14) on_embexpr_beg "\\#{"))
 (Token ((14 16) on_ident "d"))
 (Token ((14 17) on_embexpr_end "}"))
 (Token ((14 18) on_tstring_end "\""))
 (Token ((14 19) on_nl "\n"))
 (Token ((15 0) on_ignored_nl "\n"))
 (Token ((16 0) on_ident "e"))
 (Token ((16 1) on_sp " "))
 (Token ((16 2) on_op "="))
 (Token ((16 3) on_sp " "))
 (Token ((16 4) on_int "5"))
 (Token ((16 5) on_nl "\n"))
 (Token ((17 0) on_ident "e"))
 (Token ((17 1) on_sp " "))
 (Token ((17 2) on_op "**="))
 (Token ((17 5) on_sp " "))
 (Token ((17 6) on_int "2"))
 (Token ((17 7) on_nl "\n"))
 (Token ((18 0) on_ident "puts"))
 (Token ((18 4) on_sp " "))
 (Token ((18 5) on_tstring_beg "\""))
 (Token ((18 6) on_tstring_content "e: 25 == "))
 (Token ((18 15) on_embexpr_beg "\\#{"))
 (Token ((18 17) on_ident "e"))
 (Token ((18 18) on_embexpr_end "}"))
 (Token ((18 19) on_tstring_end "\""))
 (Token ((18 20) on_nl "\n"))
 (Token ((19 0) on_ignored_nl "\n"))
 (Token ((20 0) on_ident "f"))
 (Token ((20 1) on_sp " "))
 (Token ((20 2) on_op "="))
 (Token ((20 3) on_sp " "))
 (Token ((20 4) on_int "100"))
 (Token ((20 7) on_nl "\n"))
 (Token ((21 0) on_ident "f"))
 (Token ((21 1) on_sp " "))
 (Token ((21 2) on_op "<<="))
 (Token ((21 5) on_sp " "))
 (Token ((21 6) on_int "1"))
 (Token ((21 7) on_nl "\n"))
 (Token ((22 0) on_ident "puts"))
 (Token ((22 4) on_sp " "))
 (Token ((22 5) on_tstring_beg "\""))
 (Token ((22 6) on_tstring_content "f: 200 == "))
 (Token ((22 16) on_embexpr_beg "\\#{"))
 (Token ((22 18) on_ident "f"))
 (Token ((22 19) on_embexpr_end "}"))
 (Token ((22 20) on_tstring_end "\""))
 (Token ((22 21) on_nl "\n"))
 (Token ((23 0) on_ignored_nl "\n"))
 (Token ((24 0) on_ident "g"))
 (Token ((24 1) on_sp " "))
 (Token ((24 2) on_op "="))
 (Token ((24 3) on_sp " "))
 (Token ((24 4) on_int "100"))
 (Token ((24 7) on_nl "\n"))
 (Token ((25 0) on_ident "g"))
 (Token ((25 1) on_sp " "))
 (Token ((25 2) on_op ">>="))
 (Token ((25 5) on_sp " "))
 (Token ((25 6) on_int "1"))
 (Token ((25 7) on_nl "\n"))
 (Token ((26 0) on_ident "puts"))
 (Token ((26 4) on_sp " "))
 (Token ((26 5) on_tstring_beg "\""))
 (Token ((26 6) on_tstring_content "g: 50 == "))
 (Token ((26 15) on_embexpr_beg "\\#{"))
 (Token ((26 17) on_ident "g"))
 (Token ((26 18) on_embexpr_end "}"))
 (Token ((26 19) on_tstring_end "\""))
 (Token ((26 20) on_nl "\n"))
 (Token ((27 0) on_ignored_nl "\n"))
 (Token ((28 0) on_ident "h"))
 (Token ((28 1) on_sp " "))
 (Token ((28 2) on_op "="))
 (Token ((28 3) on_sp " "))
 (Token ((28 4) on_int "7"))
 (Token ((28 5) on_nl "\n"))
 (Token ((29 0) on_ident "h"))
 (Token ((29 1) on_sp " "))
 (Token ((29 2) on_op "^="))
 (Token ((29 4) on_sp " "))
 (Token ((29 5) on_int "3"))
 (Token ((29 6) on_nl "\n"))
 (Token ((30 0) on_ident "puts"))
 (Token ((30 4) on_sp " "))
 (Token ((30 5) on_tstring_beg "\""))
 (Token ((30 6) on_tstring_content "h: 4 == "))
 (Token ((30 14) on_embexpr_beg "\\#{"))
 (Token ((30 16) on_ident "h"))
 (Token ((30 17) on_embexpr_end "}"))
 (Token ((30 18) on_tstring_end "\""))
 (Token ((30 19) on_nl "\n"))
 (Token ((31 0) on_ignored_nl "\n"))
 (Token ((32 0) on_ident "i"))
 (Token ((32 1) on_sp " "))
 (Token ((32 2) on_op "="))
 (Token ((32 3) on_sp " "))
 (Token ((32 4) on_int "30"))
 (Token ((32 6) on_nl "\n"))
 (Token ((33 0) on_ident "i"))
 (Token ((33 1) on_sp " "))
 (Token ((33 2) on_op "%="))
 (Token ((33 4) on_sp " "))
 (Token ((33 5) on_int "4"))
 (Token ((33 6) on_nl "\n"))
 (Token ((34 0) on_ident "puts"))
 (Token ((34 4) on_sp " "))
 (Token ((34 5) on_tstring_beg "\""))
 (Token ((34 6) on_tstring_content "i: 2 == "))
 (Token ((34 14) on_embexpr_beg "\\#{"))
 (Token ((34 16) on_ident "i"))
 (Token ((34 17) on_embexpr_end "}"))
 (Token ((34 18) on_tstring_end "\""))
 (Token ((34 19) on_nl "\n"))
 (Token ((35 0) on_ignored_nl "\n"))
 (Token ((36 0) on_ident "j"))
 (Token ((36 1) on_sp " "))
 (Token ((36 2) on_op "="))
 (Token ((36 3) on_sp " "))
 (Token ((36 4) on_int "28"))
 (Token ((36 6) on_nl "\n"))
 (Token ((37 0) on_ident "j"))
 (Token ((37 1) on_sp " "))
 (Token ((37 2) on_op "-="))
 (Token ((37 4) on_sp " "))
 (Token ((37 5) on_int "4"))
 (Token ((37 6) on_nl "\n"))
 (Token ((38 0) on_ident "puts"))
 (Token ((38 4) on_sp " "))
 (Token ((38 5) on_tstring_beg "\""))
 (Token ((38 6) on_tstring_content "j: 24 == "))
 (Token ((38 15) on_embexpr_beg "\\#{"))
 (Token ((38 17) on_ident "j"))
 (Token ((38 18) on_embexpr_end "}"))
 (Token ((38 19) on_tstring_end "\""))
 (Token ((38 20) on_nl "\n"))
 (Token ((39 0) on_ignored_nl "\n"))
 (Token ((40 0) on_ident "k"))
 (Token ((40 1) on_sp " "))
 (Token ((40 2) on_op "="))
 (Token ((40 3) on_sp " "))
 (Token ((40 4) on_int "3"))
 (Token ((40 5) on_nl "\n"))
 (Token ((41 0) on_ident "k"))
 (Token ((41 1) on_sp " "))
 (Token ((41 2) on_op "|="))
 (Token ((41 4) on_sp " "))
 (Token ((41 5) on_int "9"))
 (Token ((41 6) on_nl "\n"))
 (Token ((42 0) on_ident "puts"))
 (Token ((42 4) on_sp " "))
 (Token ((42 5) on_tstring_beg "\""))
 (Token ((42 6) on_tstring_content "k: 11 == "))
 (Token ((42 15) on_embexpr_beg "\\#{"))
 (Token ((42 17) on_ident "k"))
 (Token ((42 18) on_embexpr_end "}"))
 (Token ((42 19) on_tstring_end "\""))
 (Token ((42 20) on_nl "\n"))
 (Token ((43 0) on_ignored_nl "\n"))
 (Token ((44 0) on_ident "l"))
 (Token ((44 1) on_sp " "))
 (Token ((44 2) on_op "="))
 (Token ((44 3) on_sp " "))
 (Token ((44 4) on_kw "false"))
 (Token ((44 9) on_nl "\n"))
 (Token ((45 0) on_ident "l"))
 (Token ((45 1) on_sp " "))
 (Token ((45 2) on_op "||="))
 (Token ((45 5) on_sp " "))
 (Token ((45 6) on_int "24"))
 (Token ((45 8) on_nl "\n"))
 (Token ((46 0) on_ident "puts"))
 (Token ((46 4) on_sp " "))
 (Token ((46 5) on_tstring_beg "\""))
 (Token ((46 6) on_tstring_content "l: 24 == "))
 (Token ((46 15) on_embexpr_beg "\\#{"))
 (Token ((46 17) on_ident "l"))
 (Token ((46 18) on_embexpr_end "}"))
 (Token ((46 19) on_tstring_end "\""))
 (Token ((46 20) on_nl "\n"))
 (Token ((47 0) on_ignored_nl "\n"))
 (Token ((48 0) on_ident "m"))
 (Token ((48 1) on_sp " "))
 (Token ((48 2) on_op "="))
 (Token ((48 3) on_sp " "))
 (Token ((48 4) on_kw "true"))
 (Token ((48 8) on_nl "\n"))
 (Token ((49 0) on_ident "m"))
 (Token ((49 1) on_sp " "))
 (Token ((49 2) on_op "&&="))
 (Token ((49 5) on_sp " "))
 (Token ((49 6) on_int "24"))
 (Token ((49 8) on_nl "\n"))
 (Token ((50 0) on_ident "puts"))
 (Token ((50 4) on_sp " "))
 (Token ((50 5) on_tstring_beg "\""))
 (Token ((50 6) on_tstring_content "m: 24 == "))
 (Token ((50 15) on_embexpr_beg "\\#{"))
 (Token ((50 17) on_ident "m"))
 (Token ((50 18) on_embexpr_end "}"))
 (Token ((50 19) on_tstring_end "\""))
 (Token ((50 20) on_nl "\n")))
